enhancements/installer/pre-flight-validations.md

```diff
@@ -0,0 +1,194 @@
+---
+title: pre-flight-validations
+authors:
+  - "@mandre"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-18
+last-updated: 2020-05-18
+status: implementable
+---
+
+# Pre-flight validations
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+- Will the validations be run automatically on `cluster create` or will it be
+  an explicit action?
+- If explicit action, how will the validations be enabled? E.g. adding
+  a `--dry-run` option to the installer vs. a separate subcommand or even
+  a separate binary.
+- When enabling the validations, will it also perform the installation or
+  simply run the validations?
+- Can the installer override validations that failed?
+- How are failures and warnings reported to the user?
+
+## Summary
+
+One of the guiding principles of OCP 4 is that the installation should always
+succeed. This goal is relatively easy to implement for public cloud platforms
+where we can make assumptions about services being available, or performance
+meeting requirements, however this is not the case with private clouds where
+each cloud is unique.
```

Comment:
Public clouds might need fewer checks, however many might apply to them as well: think quota, permissions, local connectivity (which is a requirement in some cases).

Reply:
Agree, public cloud may be of lower priority but there are many checks in place in OCM for OSD that could help bootstrap something in the AWS and GCP spaces.  There are many variables driving limits/quotas that impact if installation will be successful including support plan and region.  Explicitly checking limits up front is critical to success and we strive to tell our customers it will fail due to account/project limits before we kick off the installer.